pkg/test/resourcefixture/contexts/register.go

```diff
@@ -95,41 +92,39 @@ func GetResourceContext(fixture resourcefixture.ResourceFixture, serviceMetadata
 		rc.ResourceGVK = fixture.GVK
 	}
 
-	// Determine the type of controller for the resource: TF, DCL, or Direct.
```

Comment:
I feel ReconcilerTypeForObject() should probably be called here, and ReconcilerType should be a field under ResourceContext - it's intended to be a holder for any context related to the target test resource (the one specified in create.yaml/update.yaml).

If we want to extend the functionality to the dependent resources, it is still beneficial that we reduce the number of duplicate calls for the target resource.

Reply:
I had considered this also, but felt like it would be easier / more ergonomic for the caller of `Reconcile` (and other functions) to only have to specify the object they want reconciled. The details of figuring out which reconciler to use would be abstracted away. We could optimize a little bit by creating a cache, but I don't know that it would really have much of a performance impact, given most of the time these tests are spent waiting on GCP. It sort of feels like "premature optimization" at the expense of complicating our test code.